### PR TITLE
db turn off VerifyChecksums on reads

### PR DESCRIPTION
**Pull request by benaadams**
## Changes

- db turn off VerifyChecksums on reads; chain enforces verification (As per Besu; thanks for PR to rocksdb)

#### What types of changes does your code introduce?

- [x] Optimization

#### Requires testing

- [x] No

### CODE REVIEW

File: src/Nethermind/Nethermind.Db.Rocks/RocksDbExtensions.cs [hunk 1]
```diff
@@ -11,7 +11,14 @@ namespace Nethermind.Db.Rocks;
...
 {
-    private static readonly ReadOptions _defaultReadOptions = new();
+    private static readonly ReadOptions _defaultReadOptions = CreateReadOptions();
+
+    private static ReadOptions CreateReadOptions()
+    {
+        ReadOptions options = new();
+        options.SetVerifyChecksums(false);
+        return options;
+    }
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Db.Rocks/RocksDbExtensions.cs:21]: Make it public static on DbOnTheRocks?

**Comment by LukaszRozmej** [comment]
Do you have any performance numbers?

File: src/Nethermind/Nethermind.Db.Rocks/RocksDbExtensions.cs (hunk 1 above)
Comment (benaadams) [reply src/Nethermind/Nethermind.Db.Rocks/RocksDbExtensions.cs]: Moved

**Comment by asdacap** [comment]
Can only set for statedb?

**Comment by asdacap** [comment]
What if there is corruption, the chain does enforce verification, but that just mean that instead of database corruption error, we'll now get invalid block.

**Comment by benaadams** [comment]
> Can only set for statedb?

Feel free to mod, not too familar with the dbs

**Comment by benaadams** [comment]
> What if there is corruption, the chain does enforce verification, but that just mean that instead of database corruption error, we'll now get invalid block.

Wouldn't we get trieexception first or invalid root calculations?

**Comment by asdacap** [comment]
Invalid root == invalid block. We'll get trie exception if it can't get the node. Problem is if it get it, modifies one of its child, then recalculate and got wrong hash. So its likely invalid block.

**Comment by asdacap** [comment]
I'd get a metric first to see what are we gaining for such risk.

**Comment by asdacap** [comment]
Fun fact, if you disable checksum, and then disable compression also, you can turn on mmap read. Never tried that, could be significant.

**Comment by benaadams** [comment]
> Invalid root == invalid block. We'll get trie exception if it can't get the node. Problem is if it get it, modifies one of its child, then recalculate and got wrong hash. So its likely invalid block.

db broken eitherway; question is does it cause additional slashing risk to validators? i.e would they propose something weird?

**Comment by benaadams** [comment]
> Fun fact, if you disable checksum, and then disable compression also, you can turn on mmap read. Never tried that, could be significant.

Interesing; also saw something about `io_uring` but think that's on by default now?

**Comment by benaadams** [comment]
Doesn't seem to improve much

**Pull request closed without merging**